scripts/build_project.sh

```diff
@@ -0,0 +1,43 @@
+# Author: Teghveer Singh Ateliey
+# Date: 2024-10-30
+
+#!/bin/bash
+
+# Check args
+if [ "$#" -ne 4 ]; then
+    echo "Usage: $0 <commit-hash> <repo-path> <project> <platform>"
+    exit 1
+fi
+
+# Assign inputs to variables
+COMMIT_HASH=$1
+REPO_PATH=$2
+PROJECT=$3
+PLATFORM=$4
+
+# Checks if directory exists
+if [ ! -d "$REPO_PATH" ]; then
+    echo "Error: Repository path '$REPO_PATH' does not exist."
+    exit
+fi
+
+# Navigate to the racecar/firmware directory path
+cd "$REPO_PATH" || exit
```

Comment:
whats the `|| exit` do?

Reply:
This is meant to terminate the script if the cd command fails, in hindsight probably unnecessary since we check if the repo exists just before it.

Reply:
yeah we can safely remove that condition